### PR TITLE
PR and Code Review for HW2

### PR DESCRIPTION
**Pull request by satvika-psu**
Code changes made:
Ex-01 Added function that converts dollar amount into the equivalent number of dollars, quarters, dimes, nickels and pennies. also included test cases for error handling for invalid inputs like non-numeric values, negative values and large dollar amounts
Ex-02: Added different routes that shows its respective messages in those pages using http server and also added links to other routes in the default landing page.
ex-03 : Added different routes that shows its respective messages in its pages using Express server and also added links to other routes in the default landing page.
Ex-04: Implemented three routes here ,...one for landing page that takes to /form route......./ form route has form for the user to fill and / submit route posts the submitted details.

Feedback needed: please let me know if there is any better way to implement any of the above exercises

### CODE REVIEW

**Review by Ahmed-Al-Badri — COMMENTED** [review]
Theses files do not seem to be incorrect or missing anything, great work.

File: hw2/01-coins.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  for (let coin in coin_value) {
+    result[coin] = Math.floor(cents / coin_value[coin]);
+    cents %= coin_value[coin];
+  }
+
+  //return result;
+  return `${dollar_amt} conversion is ${JSON.stringify(result)}`;
+}
```
Comment (Ahmed-Al-Badri) [inline hw2/01-coins.js:45]: This file is somewhat hard to read, not because of the code, but because of the syntax and tabs-- they seem like they are outside of the function.

File: hw2/02-node-routing.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    res.setHeader("Cache-Control", "max-age = 86400");
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write("<h2>This Resource was cached</h2>");
+    res.end();
+  } else if (req.url === "/cookie") {
+    res.setHeader("Set-Cookie", "hello=world");
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    res.write("Cookies....Yummm");
+    res.end();
+  } else {
```
Comment (Ahmed-Al-Badri) [inline hw2/02-node-routing.js:52]: You do not really need to have else ifs, because the req.url do not change automatically.

File: hw2/03-express-routing.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    <h1>Ex- 03-Express Routing Exercise</h1>
+    <ul>
+      <li><a href="/welcome">Welcome</a></li>
+      <li><a href="/redirect">Redirect</a></li>
+      <li><a href="/cache">Cache</a></li>
+      <li><a href="/cookie">Cookie</a></li>
+      <li><a href="/other">Other</a></li>
+    </ul>
+  `);
+});
```
Comment (Ahmed-Al-Badri) [inline hw2/03-express-routing.js:37]: Nice, this part is really nice. Nice work.

File: hw2/04-form.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        <input type="text" id="username" name="username"><br>
+        <br>
+
+        <label for="email">Email:</label><br>
+        <input type="email" id="email" name="email"><br>
+        <br>
+
+        <label for="comments">Comments:</label><br>
+        <textarea id="comments" name="comments"></textarea><br>
+        <br>
```
Comment (Ahmed-Al-Badri) [inline hw2/04-form.js:40]: Added a text area for the comments is a good approach.

**Review by Ahmed-Al-Badri — APPROVED** [review]
Overall, this is great. Nice work
